istioctl/pkg/waypoint/waypoint.go

```diff
@@ -310,6 +427,9 @@ func Cmd(ctx cli.Context) *cobra.Command {
 	}
 
 	waypointApplyCmd.PersistentFlags().StringVarP(&revision, "revision", "r", "", "The revision to label the waypoint with")
+	waypointApplyCmd.PersistentFlags().BoolVarP(&waitReady, "wait-ready", "W", false, "Wait for the waypoint to be ready and synced")
+	waypointApplyCmd.PersistentFlags().DurationVar(&waitTimeout, "timeout", 90*time.Second,
+		"The duration to wait for pod to be ready and synced before failing")
```

Comment:
Note about the waiting logic: at least one execution of the readiness/synch checks is guaranteed (even if you put 0s or 0.0001 timeout), running checks are not aborted if the timeout expires while they are running. Please let me know if you would prefer the timeout to be handled differently.

Reply:
We had wait logic for the installation with `istioctl install --readiness-timeout`, but I think `--wait` is enough here without a `--timeout` for waiting a waypoint.